source/features/clone-branch.tsx

```diff
@@ -0,0 +1,89 @@
+import delegate, {DelegateEvent} from 'delegate-it';
+import gitBranch from 'octicon/git-branch.svg';
+import insertTextTextarea from 'insert-text-textarea';
+import React from 'dom-chef';
+import select from 'select-dom';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import {getRepoURL} from '../libs/utils';
+
+async function cloneBranch(event: DelegateEvent<MouseEvent, HTMLAnchorElement>): Promise<void> {
+	const currentTarget = event.delegateTarget;
+	const branchElement = currentTarget.closest<HTMLAnchorElement>('[data-branch-name]');
+	const spinner = select('.js-loading-spinner', branchElement!)!;
+	spinner.hidden = false;
+	currentTarget.hidden = true;
+
+	let newBranchName = prompt('Enter the new branch name')?.trim();
+	if (!newBranchName) {
+		return;
+	}
+
+	const getBranchInfo = await api.v3(`repos/${getRepoURL()}/git/refs/heads/${branchElement!.dataset.branchName!}`);
+	let createBranch = await createNewBranch(newBranchName, getBranchInfo);
+
+	while (Number(createBranch.httpStatus) === 422) {
+		newBranchName = prompt(String(createBranch.message) + '\n Enter the new branch name', newBranchName)!;
+		if (!newBranchName) {
+			break;
+		}
+
+		createBranch = await createNewBranch(newBranchName, getBranchInfo); // eslint-disable-line no-await-in-loop
+	}
+
+	spinner.hidden = true;
+	currentTarget.hidden = false;
+
+	if (!newBranchName) {
+		return;
+	}
+
+	const searchField = select<HTMLInputElement>('.js-branch-search-field')!;
+	searchField.select();
+	insertTextTextarea(searchField, newBranchName);
+}
+
+async function createNewBranch(newBranchName: string, getBranchInfo: AnyObject): Promise<AnyObject> {
```

Comment:
Don't pass random objects to functions.
```suggestion
async function createBranch(newBranchName: string, baseSha: string): Promise<AnyObject> {
```